### DISCUSSION TITLE
How to use GFM-mode as the default major mode for markdown files?

### DISCUSSION THREAD

**Discussion by xuhdev** [syl20bnr/spacemacs#16707] [Q&A] [answered]
I have the following line in my `dotspacemacs/user-config` configuration:

```elisp
  (add-to-list 'auto-mode-alist '("\\.\\(?:md\\|markdown\\|mkd\\|mdown\\|mkdn\\|mdwn\\)\\'" . gfm-mode))
```

However, if I start emacs to open a markdown file `emacs a.md`, Emacs always uses the normal Markdown mode. Buffers opened from within Emacs work fine.

If I move the configuration to `dotspacemacs/user-init`, markdown mode is used no matter what.

How do I use GFM mode by default?

**Comment by fnussbaum** [comment, marked as answer]
One way to achieve this would be to add 
```
(add-to-list 'major-mode-remap-alist '(markdown-mode . gfm-mode))
```
in `dotspacemacs/user-init`.

**Reply by xuhdev** [reply]
It doesn't work :(

**Reply by xuhdev** [reply]
It actually works, just need to remember to add to `dotspacemacs/user-init`. Thanks!
